File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> s;

        for(int i = 0; i < nums.size(); i++){
            int num = target - nums[i];

            if (s.count(num)){
                return {s[num],i};
            }
            s[nums[i]] = i;
        }
        return{};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        s = []
        l,r = 0,len(numbers)-1
        while l < r:
            f = numbers[l] + numbers[r]
            if f > target:
                r -= 1
            elif f < target:
                l += 1
            else:
                return[l+1,r+1]
        return s
        